exist-core/src/main/java/org/exist/backup/Backup.java

```diff
@@ -371,13 +369,33 @@ private void backup(final Set<String> seenBlobIds, final Collection current, fin
                     //Skipping resources[i]
                     continue;
                 }
+
                 resource = current.getResource(resources[i]);
 
                 if (dialog != null) {
                     dialog.setResource(resources[i]);
                     dialog.setProgress(i);
                 }
 
+                // Avoid NPE
+                if (resource == null) {
```

Comment:
So `resources` is build from `EXistCollection#listResources`.

When calling a subsequent `EXistCollection#getResource` on one of those resource names, what does a return value of `null` actually indicate? I suspect:
1. The resource has been deleted from the Collection because it was not locked during the backup process. AFAIK backup over XML:DB is best effort and not thread/concurrent safe.
2. The database has been corrupted in some way, and the resource can no longer be retrieved.

I think the warning code below should take two different paths for alerting the user depending on whether `dialog == null`. This process can run in either GUI or console mode IIRC.

Reply:
great feedback, thnx